service/controller/resource/deployment/update.go

```diff
@@ -73,49 +130,69 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		return nil, microerror.Mask(err)
 	}
 
-	if updateallowedcontext.IsUpdateAllowed(ctx) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding out which deployments have to be updated")
-
-		// Updates can be quite disruptive. We have to be very careful with updating
-		// resources that potentially imply disrupting customer workloads. We have
-		// to check the state of all deployments before we can safely go ahead with
-		// the update procedure.
-		for _, d := range currentDeployments {
-			allReplicasUp := allNumbersEqual(d.Status.AvailableReplicas, d.Status.ReadyReplicas, d.Status.Replicas, d.Status.UpdatedReplicas)
-			if !allReplicasUp {
-				r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("cannot update any deployment: deployment '%s' must have all replicas up", d.GetName()))
-				return nil, nil
-			}
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out which deployments have to be updated")
+
+	// Updates can be quite disruptive. We have to be very careful with updating
+	// resources that potentially imply disrupting customer workloads. We have
+	// to check the state of all deployments before we can safely go ahead with
+	// the update procedure.
+	for _, d := range currentDeployments {
+		allReplicasUp := allNumbersEqual(d.Status.AvailableReplicas, d.Status.ReadyReplicas, d.Status.Replicas, d.Status.UpdatedReplicas)
+		if !allReplicasUp {
+			r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("cannot update any deployment: deployment '%s' must have all replicas up", d.GetName()))
+			return nil, nil
+		}
+	}
+
+	// We select one deployment to be updated per reconciliation loop. Therefore
+	// we have to check its state on the version bundle level to see if a
+	// deployment is already up to date. We also check if there are any other
+	// changes on the pod specs. In case there are none, we check the next one.
+	// The first one not being up to date will be chosen to be updated next and
+	// the loop will be broken immediately.
+DeploymentsLoop:
```

Comment:
Missing comment?

Reply:
Nope, it's a label

Reply:
Oh - I literally have never used labels in golang before in my life :D

Reply:
Me neither before right now!